modules/actions/storedQueries.js

```diff
@@ -0,0 +1,95 @@
+import { createAction } from './utils';
+import AppConstants from '../constants';
+import { getURL } from '../../../constants/config';
+import { doGet, doPost, doDelete, doPut } from '../../utils/requestService';
+
+export function getAppStoredQueries() {
+	return (dispatch) => {
+		dispatch(createAction(AppConstants.APP.STORED_QUERIES.GET_ALL));
+		const ACC_API = getURL();
+		return doGet(`${ACC_API}/_storedqueries`)
+			.then((res) =>
+				dispatch(createAction(AppConstants.APP.STORED_QUERIES.GET_ALL_SUCCESS, res, null)),
+			)
+			.catch((error) =>
+				dispatch(createAction(AppConstants.APP.STORED_QUERIES.GET_ALL_ERROR, null, error)),
+			);
+	};
+}
+
+export function getAppStoredQuery(storedQueryId) {
+	return (dispatch) => {
+		dispatch(createAction(AppConstants.APP.STORED_QUERIES.GET));
+		const ACC_API = getURL();
+		return doGet(`${ACC_API}/_storedquery/${storedQueryId}`)
+			.then((res) => dispatch(createAction(AppConstants.APP.STORED_QUERIES.GET_SUCCESS, res)))
+			.catch((error) =>
+				dispatch(createAction(AppConstants.APP.STORED_QUERIES.GET_ERROR, null, error)),
+			);
+	};
+}
+
+export function saveAppStoredQuery(storedQueryId, payload) {
+	return (dispatch) => {
+		dispatch(createAction(AppConstants.APP.STORED_QUERIES.UPDATE));
+		const ACC_API = getURL();
+		return doPut(`${ACC_API}/_storedquery/${storedQueryId}`, payload)
+			.then((res) =>
+				dispatch(createAction(AppConstants.APP.STORED_QUERIES.UPDATE_SUCCESS, res)),
+			)
+			.catch((error) =>
+				dispatch(createAction(AppConstants.APP.STORED_QUERIES.UPDATE_ERROR, null, error)),
+			);
+	};
+}
+
+export function deleteAppStoredQuery(storedQueryId) {
+	return (dispatch) => {
+		dispatch(createAction(AppConstants.APP.STORED_QUERIES.DELETE));
+		const ACC_API = getURL();
+		return doDelete(`${ACC_API}/_storedquery/${storedQueryId}`)
+			.then((res) =>
+				dispatch(createAction(AppConstants.APP.STORED_QUERIES.DELETE_SUCCESS, res)),
+			)
+			.catch((error) =>
+				dispatch(createAction(AppConstants.APP.STORED_QUERIES.DELETE_ERROR, null, error)),
+			);
+	};
+}
+
+export function validateAppStoredQuery(storedQueryId, payload) {
+	return (dispatch) => {
+		dispatch(createAction(AppConstants.APP.STORED_QUERIES.VALIDATE));
+		const ACC_API = getURL();
+		const url = `${ACC_API}/_storedquery/${storedQueryId}/validate`;
+		return doPost(url, payload, undefined, undefined, [404, 400], true)
```

Comment:
@mohdashraf010897 Why did you ignore the `400` and `404` error codes? We want to display the error message based on it.